chore(docs): 06/01/26 docs refinements (#2235)
namehash/ensnode: docs/ensnode.io/src/content/docs/docs/integrate/why-ensnode/keep-ens-working.mdx

---
title: Keep ENS apps working 🚨

> description: A vast ecosystem of apps depends on the ENS Subgraph — ENSNode keeps them working through the ENSv2 transition.

Reviewer: ```suggestion
description: A vast ecosystem of apps depends on the ENS Subgraph — all need to be upgraded to the ENS Omnigraph to keep working in ENSv2.
```

import UpgradeToOmnigraphRequiredTable from "@components/molecules/upgrade-to-omnigraph-required-table/index.astro";
import ENSSubgraphDependentApp from "@components/molecules/upgrade-to-omnigraph-required-table/ENSSubgraphDependentApp.astro";

## The ENS Subgraph is not ENSv2 compatible

Over the years, the ENS Subgraph quietly became one of the most depended-on pieces of infrastructure in ENS. An enormous swath of the ENS ecosystem — and much of the broader web3 / Ethereum ecosystem — reads ENS data through it, directly or indirectly.

### Usage at scale

The ENS Subgraph is a cornerstone of the current ENS architecture (ENSv1), handling an extraordinary volume of requests:

- **Approx. 2 million average daily requests**
- **Over 717 million requests annually**

<figure>
  <img src="/ens-subgraph-usage.png" alt="ENS Subgraph usage metrics showing daily query volume" />
  <figcaption>
    ENS Subgraph daily query volume following the transition to Graph Network hosting (June 20, 2024
    onward)
  </figcaption>
</figure>

This volume sets the floor: it's the minimum scale at which any replacement must operate.

## Applications dependent on the ENS Subgraph

> The list below is a **partial** map of the projects and applications that depend on the ENS Subgraph for their functionality — directly, or indirectly through services like the ENS Metadata Service, ENSjs, or the Ethereum Comments Protocol. When ENSv2 launches, indexed ENS data must keep flowing to all of these, or they break.

Reviewer: ```suggestion
The list below is a **partial** map of the projects and applications that depend on the ENS Subgraph for their functionality — directly, or indirectly. When ENSv2 launches, indexed ENS data must keep flowing to all of these, or they break.
```

> ### ENS Registrars

Reviewer: <img width="1164" height="900" alt="CleanShot 2026-06-02 at 22 25 26" src="https://github.com/user-attachments/assets/5ab16a0d-f468-4282-94b7-150d13e435bf" />

Is it possible for these tooltips to open on the right side, instead of the top? That would be a lot better because currently when you open the tooltip you can't see the name of the associated app anymore!

Whatever we do here, please ensure it works well on mobile too. Thanks!

Author: On screens wider than 440px, the tooltip will appear on the right side as suggested. 

For smaller screens, such a presentation would not fit the screen, since the current implementation of the tooltip that we use here doesn't adapt to that... So for such narrow screens, I switched its placement so that the tooltip will appear below the badge. That way, it doesn't block the associated app and is fully visible for the reader.

  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://ens.tools/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>ENS Tools</strong></a>
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://snipe.zone/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Snipe Zone</strong></a>
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://github.com/blockful/nameful" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Nameful</strong></a> — by <a href="https://www.blockful.io/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Blockful</a>.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

>   <a href="https://names.linea.build/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Linea Names</strong></a> — primary management interface for <code class="bg-[var(--sl-color-bg-inline-code)] text-xs px-1.5 py-0.5 -my-0.5">linea.eth</code> subnames on Linea.

Author: Good catch, but not strictly necessary due to how little we use code. If the situation changes I'll revisit that suggestion.

  </ENSSubgraphDependentApp>
</UpgradeToOmnigraphRequiredTable>

### DAO Infrastructure

<UpgradeToOmnigraphRequiredTable>
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://snapshot.box/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Snapshot</strong></a> — via <a href="https://github.com/snapshot-labs/stamp" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Stamp</a>. This is a distinct integration from <code class="bg-[var(--sl-color-bg-inline-code)] text-xs px-1.5 py-0.5 -my-0.5">snapshot.js</code>.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://github.com/snapshot-labs/snapshot.js" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Snapshot.js</strong></a> — TypeScript library for Snapshot integration.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

>   <a href="https://daostar.org/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>DAOStar</strong></a> — by <a href="http://Metagov.org" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Metagov</a>.

Author: Applied

  <a href="https://swiss-knife.xyz/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Swiss Knife</strong></a> — by <a href="https://x.com/apoorveth" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">apoorv.eth</a>.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://wayback-machine.eth.limo/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>ENS Wayback Machine</strong></a> — by <a href="https://blossom.software/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Blossom</a>.
  </ENSSubgraphDependentApp>
</UpgradeToOmnigraphRequiredTable>

### Developer Tools

<UpgradeToOmnigraphRequiredTable>
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={true} toOmnigraphAPI={false}>
  <a href="https://www.enscribe.xyz/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Enscribe</strong></a> — ENS contract naming platform
  </ENSSubgraphDependentApp>

>   <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

Reviewer: ```suggestion
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={true} toOmnigraphAPI={false}>
```

  </ENSSubgraphDependentApp>
</UpgradeToOmnigraphRequiredTable>

### ENS Avatar & Metadata Services

<UpgradeToOmnigraphRequiredTable>
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://github.com/ensdomains/ens-metadata-service" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>ENS Metadata Service</strong></a> — the primary service used across the ENS ecosystem for loading ENS avatar images.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://github.com/snapshot-labs/stamp" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Stamp</strong></a> — by <a href="https://snapshot.box/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Snapshot Labs</a>.
  </ENSSubgraphDependentApp>

>   <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

Reviewer: ```suggestion
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={true} toOmnigraphAPI={false}>
```

  <a href="https://github.com/ethereumidentitykit/ens-metadata-flarecloud" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>ENS Metadata Flarecloud</strong></a> — by the <a href="https://ethid.org/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">EthId Foundation</a>.
  </ENSSubgraphDependentApp>
</UpgradeToOmnigraphRequiredTable>

### Social Applications

<UpgradeToOmnigraphRequiredTable>

>   <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

Reviewer: ```suggestion
  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={true} toOmnigraphAPI={false}>
```

  <a href="https://efp.app/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Ethereum Follow Protocol</strong></a> — onchain social graph protocol.
  </ENSSubgraphDependentApp>

>   <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

Reviewer: <img width="1530" height="1350" alt="CleanShot 2026-06-02 at 22 32 45" src="https://github.com/user-attachments/assets/fa4f0d3b-ed60-4227-8991-fa0382ab7bda" />

Please move each of these to the bottom of the "Social Applications" category.

  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://decentraland.org/create" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Decentraland Creator Hub</strong></a> (<a href="https://github.com/decentraland/creator-hub" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">GitHub</a>) — by <a href="https://decentraland.org/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Decentraland</a>.
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>
  <a href="https://next.id/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Next.ID</strong></a>
  </ENSSubgraphDependentApp>

  <ENSSubgraphDependentApp toENSNodeSubgraphCompatibleEndpoint={false} toOmnigraphAPI={false}>

>   <a href="http://web3.bio/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content"><strong>Web3.bio</strong></a> — builds on <a href="https://next.id/" target="_blank" rel="noopener noreferrer" class="omnigraph-required-table-content">Next.ID</a>.

Author: Applied

  </ENSSubgraphDependentApp>
</UpgradeToOmnigraphRequiredTable>

:::note[Help us expand this list]
Do you know of other applications that rely on the ENS Subgraph that aren't listed here? Please [open a PR](https://github.com/namehash/ensnode) to update this documentation.
:::

## What this means

The breadth of applications above shows that indexed ENS data is critical infrastructure for ENS and the broader ecosystem — and that there is large, proven demand for it today. When ENSv2 launches, the Subgraph's [Key Limitations](/docs/integrate/ens-subgraph/key-limitations) become breaking for these apps. ENSNode exists to keep indexed ENS data flowing through the transition and beyond, with a robust, scalable, multichain, ENSv2-ready replacement.

## Already building directly on ENSNode

A growing set of companies and apps have already moved onto ENSNode — [see who's already building on ENSNode](/docs/integrate/why-ensnode#join-those-already-building-on-ensnode).

## Next Steps

Next, consider the key limitations of the legacy ENS Subgraph.

<LinkCard
  href="/docs/integrate/ens-subgraph/key-limitations"

>   title="Key Limitations"

Reviewer: ```suggestion
  title="Key Limitations of the Subgraph 🚨"
```
